R/maic_unanchored.R

```diff
@@ -131,107 +147,269 @@ maic_unanchored <- function(weights_object,
+    report_table_tte(coxobj_dat_adj, medSurv_dat_adj, tag = paste0("After matching/", endpoint_name))
+  )
+
+  if (is.null(res$inferential[["boot_est"]])) {
+    res$inferential[["report_overall_bootCI"]] <- NULL
+  } else {
+    boot_res_AB <- res_AB
+    boot_logres_se <- sd(log(res$inferential[["boot_est"]]), na.rm = TRUE)
+    if (boot_ci_is_quantile) {
+      boot_res_AB$ci_l <- quantile(res$inferential[["boot_est"]], p = 0.025)
+      boot_res_AB$ci_u <- quantile(res$inferential[["boot_est"]], p = 0.975)
     } else {
-      res$inferential[["boot_est"]] <- NULL
+      boot_res_AB$ci_l <- exp(log(boot_res_AB$est) + qnorm(0.025) * boot_logres_se)
+      boot_res_AB$ci_u <- exp(log(boot_res_AB$est) + qnorm(0.975) * boot_logres_se)
     }
-
-    # make analysis report table
-    res$inferential[["report_overall_robustCI"]] <- rbind(
-      report_table(coxobj_dat, medSurv_dat, tag = paste0("Before matching/", endpoint_name)),
-      report_table(coxobj_dat_adj, medSurv_dat_adj, tag = paste0("After matching/", endpoint_name))
+    tmp_report_table_tte <- report_table_tte(coxobj_dat_adj, medSurv_dat_adj, tag = paste0("After matching/", endpoint_name))
+    tmp_report_table_tte$`HR[95% CI]`[1] <- paste0(
+      format(round(boot_res_AB$est, 2), nsmall = 2), "[",
+      format(round(boot_res_AB$ci_l, 2), nsmall = 2), ";",
+      format(round(boot_res_AB$ci_u, 2), nsmall = 2), "]"
     )
+    res$inferential[["report_overall_bootCI"]] <- rbind(
+      report_table_tte(coxobj_dat, medSurv_dat, tag = paste0("Before matching/", endpoint_name)),
+      tmp_report_table_tte
+    )
+  }
+
+  # output
+  res
+}
+
+# MAIC inference functions for Binary outcome type ------------
+
+maic_unanchored_binary <- function(res,
+                                   res_AB,
+                                   dat,
+                                   ipd,
+                                   pseudo_ipd,
+                                   binary_robust_cov_type,
+                                   weights_object,
+                                   endpoint_name,
+                                   eff_measure,
+                                   boot_ci_is_quantile,
+                                   trt_ipd,
+                                   trt_agd) {
+  # ~~~ Analysis table
+  # : set up proper link
+  glm_link <- switch(eff_measure,
+    "RD" = poisson(link = "identity"),
+    "RR" = poisson(link = "log"),
+    "OR" = binomial(link = "logit")
+  )
+
+  # : fit glm for binary outcome and robust estimate with weights
+  binobj_dat <- glm(RESPONSE ~ ARM, dat, family = glm_link)
+  binobj_dat_adj <- glm(RESPONSE ~ ARM, dat, weights = weights, family = glm_link)
+  bin_robust_cov <- clubSandwich::vcovCR(binobj_dat_adj,
+    cluster = dat$USUBJID,
+    type = binary_robust_cov_type
+  )
+  bin_robust_coef <- clubSandwich::conf_int(binobj_dat_adj, bin_robust_cov, coef = 2, p_values = TRUE)
```

Comment:
I think we should use sandwich package and not clubsandwich
let's discuss

Reply:
@MikeJSeo Which sandwich `vcov*()` could we use?

Reply:
switched to sandwich wtih HC3 by default. Using lmtest to get CIs